tests/test_conversation_utils.py

```diff
@@ -0,0 +1,75 @@
+from khoj.processor.conversation import utils
+from langchain.schema import ChatMessage
+import factory
+import tiktoken
+
+
+class ChatMessageFactory(factory.Factory):
+    class Meta:
+        model = ChatMessage
+
+    content = factory.Faker("paragraph")
+    role = factory.Faker("name")
+
+
+class TestTruncateMessage:
+    max_prompt_size = 4096
+    model_name = "gpt-3.5-turbo"
+    encoder = tiktoken.encoding_for_model(model_name)
+
+    def test_truncate_message_all_small(self):
+        chat_messages = ChatMessageFactory.build_batch(500)
+        assert len(chat_messages) == 500
```

Comment:
nit: this assert seems unnecessary as it's testing the factory package rather than unit testing khoj code

Reply:
True, fair point. Let me remove it.